pkg/jxfactory/factory.go

```diff
@@ -0,0 +1,183 @@
+package jxfactory
```

Comment:
I like the idea of splitting out the jxfactory. This will make it also easier for Apps which need a client to modify Jenkins-X CRDs. That said, I think we should adjust the whole codebase in this case opposed to duplicate code.

Reply:
as commented elsewhere in the review; the code in clients is dependent on CLI flags + a static variable + works a little different; so its probably better for now to keep both factory implementations and over time try remove the old factory with the new factory where we can